tensorflow/lite/experimental/micro/kernels/sub.cc

```diff
@@ -0,0 +1,492 @@
+/* Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow/lite/c/builtin_op_data.h"
+#include "tensorflow/lite/c/common.h"
+#include "tensorflow/lite/kernels/internal/common.h"
+#include "tensorflow/lite/kernels/internal/quantization_util.h"
+#include "tensorflow/lite/kernels/internal/reference/process_broadcast_shapes.h"
+#include "tensorflow/lite/kernels/internal/tensor_ctypes.h"
+#include "tensorflow/lite/kernels/kernel_util.h"
+#include "tensorflow/lite/kernels/op_macros.h"
+
+namespace tflite {
+namespace ops {
+namespace micro {
+namespace sub {
+
+namespace {
+
```

Comment:
Can you put these math functions in a new file in tensorflow/lite/kernels/internal/reference/sub.h, remove the duplicate code from tensorflow/lite/kernels/internal/reference/reference_ops.h, and #include the file at the top of reference_ops.h?

That will help us use the same code from TFL mobile and micro, so we aren't increasing the duplication.

Reply:
Hi Peter, 
The reason I didn't do this, is because there is a ScopedProfilingLabelWrapper definition in (most of) the reference_ops.h reference kernels which does not compile in micro. Do you have any suggestion about how can I fix it? Should I remove the definition?

Thanks

Reply:
Hi Pete, 
I was wondering if you had any update about this PR. 

Thank you a lot

Reply:
We recently added ruy::profiler::ScopeLabel label(); from  tensorflow/lite/experimental/ruy/profiler/instrumentation.h.  This #ifdefs this label to a no-op if RUY_PROFILER is not defined, which makes it work on micro.

Please create tensorflow/lite/kernels/internal/reference/sub.h like Pete mentioned, but use ruy::profiler::ScopeLabel label();